scanamo/src/main/scala/org/scanamo/query/ConditionExpression.scala

```diff
@@ -38,11 +40,11 @@ case class ConditionalOperation[V, T](tableName: String, t: T)(
       None
     )
     ScanamoOps
-      .conditionalUpdate(unconditionalRequest.copy(condition = Some(state.apply(t)(unconditionalRequest.condition))))
+      .update(unconditionalRequest.copy(condition = Some(state.apply(t)(unconditionalRequest.condition))))
       .map(
         either =>
           either
-            .leftMap[ScanamoError](ConditionNotMet(_))
+            .leftMap[ScanamoError](DynamoDBException(_))
```

Comment:
@regiskuckaertz & @TBonnin It would be good to get some feedback on this specific change. Switching from `conditionalUpdate` to `update` has made the type of the Left in the Either more general. Does it matter that there is not more granularity in the error?

Reply:
Hmm, this won't be that easy. We don't want condition checks to break the whole computation, but we cannot say that for other errors. We may want to add a new constructor to `ScanamoOpsA`:
```scala
final case class Fail(t: Throwable) extends ScanamoOpsA[Nothing]
```
and make `ScanamoOps` and instance of `MonadError`:
```scala
implicit val monadErrorScanamoOps = new MonadError[ScanamoOps, Throwable] {
  def raiseError[A](e: Throwable) = liftF[ScanamoOpsA, A](Fail(e))

  def handleErrorWith[A](fa: ScanamoOps[A])(f: Throwable => ScanamoOps[A]) =
    fa resume match {
      case Left(Fail(t)) => f(t)
      case Left(sfa)     => liftF(sfa)
      case Right(a)      => a.pure
    }
}
```
then we can put errors back on the stack for someone to deal with:
```scala
ScanamoOps.update(...)
  .flatMap {
    case Left(t) if t.isInstanceOf[ConditionCheckFailedException] => Left(ConditionNotMet(t)).pure
    case Left(t) => raiseError(t)
    case Right(r) => Right(r).pure
  }
```